app/controllers/green_lanes/results_controller.rb

```diff
@@ -12,20 +12,24 @@ def create
       @country_of_origin = results_params[:country_of_origin] || GeographicalArea::ERGA_OMNES
       @moving_date = results_params[:moving_date]
       @category = category
+      # TODO: THIS IS STILL NOT WORKING!!!!!
       @answers = JSON.parse(results_params[:ans].presence || '{}')
       @assessments = AssessmentsPresenter.new(determine_category, @answers)
     end
 
     private
 
     def results_params
+      parsed_ans = JSON.parse(params[:ans]) if params[:ans].present? && !params.is_a?(ActionController::Parameters)
+      params[:ans] = parsed_ans if parsed_ans
+
       params.permit(
         :commodity_code,
         :country_of_origin,
         :moving_date,
         :c1ex,
         :c2ex,
-        ans: {},
+        ans: permit_dynamic_ans(parsed_ans),
```

Comment:
we started using `answers` instead of `ans` to make it clear that it is plural.
(if you are sure it is singular you can use `ans`)